File: src/bin/day1_2.rs
use std::fs::read_to_string;

const IS_PROD: bool = true;

const TEST_INPUT_PATH: &str = "./inputs/day1.test.in";
const PROD_INPUT_PATH: &str = "./inputs/day1.prod.in";
const INPUT_PATH: &str = if IS_PROD {
    PROD_INPUT_PATH
} else {
    TEST_INPUT_PATH
};

fn main() {
    let mut a: Vec<usize> = vec![];
    let mut b: Vec<usize> = vec![];

    read_to_string(INPUT_PATH)
        .unwrap()
        .trim()
        .split("\n")
        .for_each(|l| {
            let values = l
                .split("   ")
                .map(|v| v.parse::<usize>().unwrap())
                .collect::<Vec<usize>>();

            a.push(*values.get(0).unwrap());
            b.push(*values.get(1).unwrap());
        });

    let mut result = 0;
    let vec_len = a.len();

    for n in 0..vec_len {
        let va = a.get(n).unwrap();
        let mut repeats = 0;

        for x in 0..vec_len {
            let vb = b.get(x).unwrap();
            if va == vb {
                repeats += 1;
            }
        }

        result += va * repeats;
    }

    println!("Result : {:?}", result);
}

File: src/day1_2.ts
const IS_PROD = true;
const TEST_INPUT_PATH = "./inputs/day1.test.in";
const PROD_INPUT_PATH = "./inputs/day1.prod.in";
const INPUT_PATH = IS_PROD ? PROD_INPUT_PATH : TEST_INPUT_PATH;

const file = Bun.file(INPUT_PATH);
const text = await file.text();

const a: number[] = [];
const b: number[] = [];

text.trim().split("\n").forEach((line) => {
  const [va, vb] = line.split("   ");
  a.push(Number.parseInt(va, 10));
  b.push(Number.parseInt(vb, 10));
});

let result = 0;

for (let i = 0; i < a.length; i += 1) {
  let repeats = 0;

  for (let j = 0; j < b.length; j += 1) {
    if (a[i] === b[j]) {
      repeats += 1;
    }
  }

  result += a[i] * repeats;
}

console.log(result);
